src/libraries/Microsoft.PowerFx.Core/Binding/BinderCheckTypeResult.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Binding
+{
+    internal class BinderCheckTypeResult
```

Comment:
Could these reasonably be `readonly struct`? We haven't really been using this yet in PowerFx, but this seems suited to it. 
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct#readonly-struct

Reply:
Done